test/middleware/CompanyAuthMiddleware.test.ts

```diff
@@ -156,13 +161,57 @@ describe('Company Authentication Middleware', () => {
 
         const startingWithRedirectUrl = (redirect : string) => redirect.startsWith(redirectUrl);
 
-        await companyAuthMiddleware.handler(request, response, nextFunction);
+        await companyAuthMiddleware.handler(req, response, nextFunction);
         nextFunction.didNotReceive();
         response.received(1).redirect(Arg.is(startingWithRedirectUrl));
 
     });
 });
 
+describe('All pages after the Penalty Details page:', () => {
+
+    const pageList = [
+        ['Select Penalty', SELECT_THE_PENALTY_PAGE_URI],
+        ['Review Penalty', REVIEW_PENALTY_PAGE_URI]
+    ];
+
+    pageList.forEach((page) => {
+        it(`The ${page[0]} page should redirect unauthenticated user to the Company Auth service`, async () => {
+
+            const applicationData: Partial<ApplicationData> = {
+                appeal: {
+                    penaltyIdentifier: {
+                        companyNumber: 'NI999999',
+                        penaltyReference: 'PEN1A/NI000000',
+                        companyName: 'Test',
+                        penaltyList: {
+                            items: [
+                                {} as Penalty,
+                                {} as Penalty
```

Comment:
I guess number of items does not matter now so is that even needed? Shall we remove company name and penalty list from session data?

Reply:
You're right. I tried removing the penalty list a while ago but the tests were failing so I left them in. Now that I've fixed the middleware order I'm able to remove the surplus session data.